src/main/java/bio/terra/workspace/service/resource/controlled/WsmResourceType.java

```diff
@@ -0,0 +1,20 @@
+package bio.terra.workspace.service.resource.controlled;
+
+import bio.terra.workspace.service.datareference.model.DataReferenceType;
+
+// TODO: should this merge with DataReferenceType? References and controlled resources support
+// different types.
+public enum WsmResourceType {
```

Comment:
I think we should have one enumeration for resource type and it should be cloud-specific. Buckets are the tiny exception where an object is fairly common. I think it will actually make code more complex if we have to check two enumerations to figure out what thing we are dealing with.

Reply:
We should maintain the cloud type enum. That is useful on its own and the revised schema needs that. I suggest we put the cloud type as a field on the resourceType enum so we can carry it as a single thing.

Reply:
I like that solution. I didn't make an internal cloud platform enum yet; it's just using the API one. Is that something we need?